daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -745,7 +746,9 @@ object Ast {
       choices: Map[ChoiceName, GenTemplateChoice[E]], // Choices available in the template.
       observers: E, // Observers of the contract.
       key: Option[GenTemplateKey[E]],
-      implements: Map[TypeConName, GenTemplateImplements[E]],
+      implements: ListMap[TypeConName, GenTemplateImplements[
```

Comment:
Unfortunately `ListMap` has bad complexity. See [Scala Std Library Performance Characteristics](https://docs.scala-lang.org/overviews/collections/performance-characteristics.html). 
I think we have to store the order as a separate field, or revive `InsertOrderMap` .

Reply:
I think I’d go for restoring InsertOrderMap. Seems easier to maintain than to keep the order in a separate field.

Reply:
Let's do that in another PR.

Reply:
I replaced ListMap with VectorMap, which should give us amortized O(1) performance on the lookups and preserves insertion order. An alternative that also preserves insertion order is TreeSeqMap.

Reply:
Good finding. I was not aware of this addition to scala 2.13. 
Thanks.